tests/func/test_delta.py

```diff
@@ -0,0 +1,286 @@
+import os
+
+import pytest
+import regex as re
+from PIL import Image
+
+import datachain as dc
+from datachain import func
+from datachain.error import DatasetVersionNotFoundError
+from datachain.lib.dc import C
+from datachain.lib.file import File, ImageFile
+
+
+def test_delta_update_from_dataset(test_session, tmp_dir, tmp_path):
```

Comment:
we should have a test with map and we should count number of maps applied in a regular mode and in delta (to make sure we don't do things twice)